### ISSUE TITLE
CNN.Model pruning: no gain in speeding up of inference

### ISSUE THREAD

**Issue by NikishinRoman** [tensorflow/tensorflow#22732]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Windows 7
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:N/A
- **TensorFlow installed from (source or binary)**:pip install
- **TensorFlow version (use command below)**:1.10.1
- **Python version**:3.6
- **Bazel version (if compiling from source)**:N/A
- **GCC/Compiler version (if compiling from source)**:N/A
- **CUDA/cuDNN version**:use CPU
- **GPU model and memory**:N/A
- **Exact command to reproduce**:N/A

I implemented iterative pruning of a convolutional neural network using the example tensorflow **/tensorflow/contrib/model_pruning/** in order to increase the speed of inference .
The result of the procedure is an increase in the number of zero weight values(increase sparcity), which are still stored in dense tensors. I do not see a gain in speed.
How can I eliminate null weights used in convolutional layers from calculations? Can I remove  conv  filter knowing its sparseness?
Please give some recomendation on these issues.

**Comment by ymodak** [comment]
The example model works as expected. Its difficult to decipher why you aren't observing gain in speed. Some general guidelines to solve your issues:
You can try adding dropout layer to reduce sparsity.
I wouldn't recommend removing conv layer since dropout should help in regularization.
Also I would like to recommend referring this [research paper](https://openreview.net/pdf?id=SkC_7v5gx)
It talks about how does to design a network that balances its size, training time, and prediction accuracy.

**Comment by suyoggupta** [comment]
As noted here: https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/model_pruning#removing-pruning-ops-from-the-trained-graph-
The model_pruning library helps with the training of a sparse model. It is left upto the underlying hardware platform and inference engine to implement sparse ops to provide acceleration

**Comment by ymodak** [comment]
Closing this issue since Suyog's explanation throws light on the speed aspect related to CNN Model pruning. Feel free to reopen this issue if you get any additional information that we haven't considered.

**Comment by ymodak** [comment]
Are you satisfied with the resolution of your issue?
[Yes](https://goo.gl/forms/Oe0tEvODFRoI2gJF3)
[No](https://goo.gl/forms/fUjzOfrtkFbrOT8d2)
